50_wa_diff_overdose.py

```diff
@@ -0,0 +1,299 @@
+deaths = pd.read_csv("05_cleaned_data/deaths_wa.csv")
+# two separate dfs - one for washington and one for comp states
+washington = deaths[deaths["StateName"] == "Washington"]
+# comp = deaths[deaths["StateName"] != "Washington"]
+comp = deaths[
+    (deaths["StateName"] == "Oregon")
+    | (deaths["StateName"] == "Idaho")
+    | (deaths["StateName"] == "Montana")
+    | (deaths["StateName"] == "Nevada")
+    | (deaths["StateName"] == "Wyoming")
+]
+
+
+def select_overdose(record):
+    """Simple function to select only overdose records"""
+
+    if record == "All other non-drug and non-alcohol causes":
+        return 0
+
+    if record == "All other alcohol-induced causes":
+        return 0
+
+    if record == "All other drug-induced causes":
+        return 0
+
+    if record == "Alcohol poisonings (overdose) (X45, X65, Y15)":
+        return 0
+
+    if record == "Drug poisonings (overdose) Unintentional (X40-X44)":
+        return 1
+
+    if record == "Drug poisonings (overdose) Suicide (X60-X64)":
+        return 1
+
+    if record == "Drug poisonings (overdose) Undetermined (Y10-Y14)":
+        return 1
+
+    else:
+        return "error"
+
+
+# copy to fix the dreaded "A value is trying to be set on a copy of a slice" error
+wa_deaths = washington.copy()
+comp_deaths = comp.copy()
+
+# apply new function to our df
+wa_deaths["overdose"] = wa_deaths["Drug/Alcohol Induced Cause"].apply(
+    lambda x: select_overdose(x)
+)
+comp_deaths["overdose"] = comp_deaths["Drug/Alcohol Induced Cause"].apply(
+    lambda x: select_overdose(x)
+)
+
+# filter accordingly based on new column
+wa_deaths = wa_deaths[wa_deaths["overdose"] != 0]
+comp_deaths = comp_deaths[comp_deaths["overdose"] != 0]
+
+wa_deaths["overdose_per_100k"] = wa_deaths["Deaths"] / wa_deaths["Population"] * 100_000
+comp_deaths["overdose_per_100k"] = (
+    comp_deaths["Deaths"] / comp_deaths["Population"] * 100_000
+)
+
+wa_result = (
+    wa_deaths.groupby(["Year", "County"])["overdose_per_100k"].sum().reset_index()
+)
+comp_result = (
+    comp_deaths.groupby(["Year", "StateName", "CountyName"])["overdose_per_100k"]
+    .sum()
+    .reset_index()
+)
+comp_result[comp_result["StateName"] == "Oregon"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Idaho"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Montana"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Nevada"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Wyoming"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+wa_result.groupby("Year")["overdose_per_100k"].mean()
+comp_result.groupby("Year")["overdose_per_100k"].mean()
+wa_result = wa_result.groupby("Year")["overdose_per_100k"].mean().reset_index()
+
+comp_result = comp_result.groupby(["Year"])["overdose_per_100k"].mean().reset_index()
```

Comment:
The code looks a bit redundant here. I mean you could have directly done like this (as shown below) in lines 96 and 97. Then lines 98 and 100 were not required. 

wa_result = wa_result.groupby(["Year"])["overdose_per_100k"].mean().reset_index()
comp_result = comp_result.groupby(["Year"])["overdose_per_100k"].mean().reset_index()

Reply:
Yes. I made stepwise group by to check what my data looks like with certain collapse.